spec/lib/tasks/local_custodian_codes_from_csv_spec.rb

```diff
@@ -0,0 +1,10 @@
+describe "Import local custodian codes task" do
+  describe "import:local_custodian_codes" do
+    it "should update local_custodian_code field for LocalAuthority with correct code" do
+      args = Rake::TaskArguments.new(%i[filename], ["spec/lib/local-links-manager/import/fixtures/addressbase-local-custodian-codes-test.csv"])
+      local_authority = create(:local_authority, slug: "leicestershire", local_custodian_code: nil)
+      expect { Rake::Task["import:local_custodian_codes"].execute(args) }
+        .to change { local_authority.reload.local_custodian_code }.from(nil).to("2460")
```

Comment:
This is kind of a repeat of the unit tests in local_custodian_codes_importer_spec.rb. We've already tested the functionality extensively in there, so this test can simply check that a new instance of `LocalLinksManager::Import::LocalCustodianCodesImporter` was created.

It might be tricky to do that without using `allow_any_instance_of` - [some ideas](https://github.com/alphagov/locations-api/pull/14#discussion_r732869828) here.

Reply:
Thank you for looking at it @ChrisBAshton.  In this case, I think it's better to check the actual change than mocks. The test looks the same, but it has a different role. This is more a feature/acceptance spec, as it checks one of the application entrypoints, while test for LocalCustodianCodeImporter is a simple unit spec. I would avoid to use mocks when they are not necessary, leaving them mostly for unit specs, where they help test classes in isolation, like in the PR you linked.

Reply:
Fair enough - thanks for looking 👍